fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -234,24 +234,24 @@ public static Object toClassWithRealEquals(@Nullable Object obj) {
 
     @SuppressWarnings("unchecked")
     public static int compare(@Nullable Object fieldValue, @Nullable Object comparand) {
-        if (fieldValue == null) {
-            if (comparand == null) {
-                return 0;
-            } else {
-                return -1;
-            }
-        } else if (comparand == null) {
-            return 1;
+        return toComparable(fieldValue).compareTo(toComparable(comparand));
+    }
+
+    @SpotBugsSuppressWarnings("NP_BOOLEAN_RETURN_NULL")
+    private static boolean compareEquals(@Nonnull Object value, @Nonnull Object comparand) {
+        if (value instanceof Message) {
+            return MessageHelpers.compareMessageEquals(value, comparand);
         } else {
-            return toComparable(fieldValue).compareTo(toComparable(comparand));
+            return toClassWithRealEquals(value).equals(toClassWithRealEquals(comparand));
```

Comment:
Can we make `toClassWithRealEquals` also `@Nonnull` and remove the `null` check in it, to further ensure that all callers are similarly consistent with the rest of this change?

Reply:
Done.